src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,187 @@
+package parser;
+
+import exception.DukeException;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Arrays;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import task.Constant;
+
+public class Parser {
+    /**
+     * @param input raw input
+     * @return Boolean returns true if input is a done/delete command
+     * @throws DukeException if user input matches a done/delete command but it not properly
+     *     formatted
+     */
+    private static String[] timeRegex = {
+        "H:m", "Hmm", "H.m", "h.m a", "h.m a", "h:m a", "h:m a", "hmm a", "hmma", "h a", "ha", "H"
+    };
+
+    private static String[] dateRegex = {
+        "ddMMyyyy", "yyyyMMdd", "d-M-yyyy", "d/M/yyyy", "yyyy-M-d", "yyyy/M/d"
+    };
+
+    public static Boolean isDoneDelete(String input) throws DukeException {
```

Comment:
Method should be named `isDoneOrDelete()`. Shortening method names can often confuse the reader, so its better to use a longer name in case where the name is still a reasonable length.